### ISSUE TITLE
Crash on device startup

### ISSUE THREAD

**Issue by mariobrostech** [ImranR98/Obtainium#1221]
**Describe the bug**
The Android System displays a crash report for Obtainium shortly following the device startup sequence.

**To Reproduce**
1. Install Obtainium
2. Add your preferred apps and sources to the Obtainium app and enable automatic background updates
3. Reboot the device
4. Observe the app crash and show the attached crash report

**Screenshots and Logs**
```
type: crash
osVersion: google/panther/panther:14/UQ1A.231205.015/2023121200:user/release-keys
package: dev.imranr.obtainium:2238
process: dev.imranr.obtainium
processUptime: 107 + 322 ms
installer: dev.imranr.obtainium

java.lang.RuntimeException: Unable to instantiate receiver dev.fluttercommunity.plus.androidalarmmanager.RebootBroadcastReceiver: java.lang.ClassNotFoundException: Didn't find class "dev.fluttercommunity.plus.androidalarmmanager.RebootBroadcastReceiver" on path: DexPathList[[zip file "/data/app/~~P88G99kM3nTfWzRoT1I84A==/dev.imranr.obtainium-QKo05j0lJCHYEqx7ywvqhA==/base.apk"],nativeLibraryDirectories=[/data/app/~~P88G99kM3nTfWzRoT1I84A==/dev.imranr.obtainium-QKo05j0lJCHYEqx7ywvqhA==/lib/arm64, /data/app/~~P88G99kM3nTfWzRoT1I84A==/dev.imranr.obtainium-QKo05j0lJCHYEqx7ywvqhA==/base.apk!/lib/arm64-v8a, /system/lib64, /system_ext/lib64]]
	at android.app.ActivityThread.handleReceiver(ActivityThread.java:4466)
	at android.app.ActivityThread.-$$Nest$mhandleReceiver(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2268)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.app.ActivityThread.main(ActivityThread.java:8216)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ExecInit.main(ExecInit.java:49)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:359)
Caused by: java.lang.ClassNotFoundException: Didn't find class "dev.fluttercommunity.plus.androidalarmmanager.RebootBroadcastReceiver" on path: DexPathList[[zip file "/data/app/~~P88G99kM3nTfWzRoT1I84A==/dev.imranr.obtainium-QKo05j0lJCHYEqx7ywvqhA==/base.apk"],nativeLibraryDirectories=[/data/app/~~P88G99kM3nTfWzRoT1I84A==/dev.imranr.obtainium-QKo05j0lJCHYEqx7ywvqhA==/lib/arm64, /data/app/~~P88G99kM3nTfWzRoT1I84A==/dev.imranr.obtainium-QKo05j0lJCHYEqx7ywvqhA==/base.apk!/lib/arm64-v8a, /system/lib64, /system_ext/lib64]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:259)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
	at android.app.AppComponentFactory.instantiateReceiver(AppComponentFactory.java:110)
	at androidx.core.app.CoreComponentFactory.instantiateReceiver(Unknown Source:0)
	at android.app.ActivityThread.handleReceiver(ActivityThread.java:4459)
	... 11 more

```

**Please complete the following information:**
 - Device: Pixel 7
 - OS: Android 14
 - Obtainium Version: v0.15.2-beta

**Comment by ImranR98** [comment]
Did this happen only once or on every startup?

**Comment by mariobrostech** [comment]
This happens every time that my device starts up. One thing that might influence this behavior is that I have `Always show crash dialog` and `Show background ANRs` enabled in my device's developer options. However, I do recall seeing this report rarely occur even before I had enabled these toggles in Developer Options.

**Comment by ImranR98** [comment]
Could you try exporting your data (with settings), uninstalling the app, then reinstalling/restoring?

The error refers to a class that no longer exists because it is part of a Flutter plugin that was removed in version 0.15.0. For some reason Android still tries to use it, I'm hoping a reinstall will fix this.

**Comment by mariobrostech** [comment]
Sure, I will give that a try and report back with my findings. Please note that since this is a device that I rely on for my work, it may take a few days to confirm that the crash no longer occurs. If you don't hear from me in the next few days, please feel free to reply and bump this issue. Thanks for your help!

**Comment by 25huizengek1** [comment]
Having the same issues, but actually in all cases where Obtainium is in the background. I think this is caused by the fact that the receivers for these classes are still registered in the manifest (https://github.com/ImranR98/Obtainium/blob/main/android%2Fapp%2Fsrc%2Fmain%2FAndroidManifest.xml#L51-L61), even though the dependency is not specified anymore (changed here: https://github.com/ImranR98/Obtainium/commit/77d81716ed34afea12fb6df6955847833bcd873e#diff-8b7e9df87668ffa6a04b32e1769a33434999e54ae081c52e5d943c541d4c0d25L60)

Not entirely sure if this fixes everything though, but I think these classes should at least not be referenced anymore since they do not exist on runtime.

The app continuously crashes every ~3 hours:
```FATAL EXCEPTION: main
Process: dev.imranr.obtainium, PID: 20937
java.lang.RuntimeException: Unable to instantiate receiver dev.fluttercommunity.plus.androidalarmmanager.AlarmBroadcastReceiver: java.lang.ClassNotFoundException: Didn't find class "dev.fluttercommunity.plus.androidalarmmanager.AlarmBroadcastReceiver" on path: DexPathList[[zip file "/data/app/~~68jUKHdwuHqwdJm7fq14EA==/dev.imranr.obtainium-xvYdiiP9IVrLT_0r8HVMBA==/base.apk"],nativeLibraryDirectories=[/data/app/~~68jUKHdwuHqwdJm7fq14EA==/dev.imranr.obtainium-xvYdiiP9IVrLT_0r8HVMBA==/lib/arm64, /data/app/~~68jUKHdwuHqwdJm7fq14EA==/dev.imranr.obtainium-xvYdiiP9IVrLT_0r8HVMBA==/base.apk!/lib/arm64-v8a, /system/lib64, /system/system_ext/lib64]]
	at android.app.ActivityThread.handleReceiver(ActivityThread.java:4859)
	at android.app.ActivityThread.-$$Nest$mhandleReceiver(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2420)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:226)
	at android.os.Looper.loop(Looper.java:313)
	at android.app.ActivityThread.main(ActivityThread.java:8762)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:604)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
Caused by: java.lang.ClassNotFoundException: Didn't find class "dev.fluttercommunity.plus.androidalarmmanager.AlarmBroadcastReceiver" on path: DexPathList[[zip file "/data/app/~~68jUKHdwuHqwdJm7fq14EA==/dev.imranr.obtainium-xvYdiiP9IVrLT_0r8HVMBA==/base.apk"],nativeLibraryDirectories=[/data/app/~~68jUKHdwuHqwdJm7fq14EA==/dev.imranr.obtainium-xvYdiiP9IVrLT_0r8HVMBA==/lib/arm64, /data/app/~~68jUKHdwuHqwdJm7fq14EA==/dev.imranr.obtainium-xvYdiiP9IVrLT_0r8HVMBA==/base.apk!/lib/arm64-v8a, /system/lib64, /system/system_ext/lib64]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:259)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
	at android.app.AppComponentFactory.instantiateReceiver(AppComponentFactory.java:110)
	at androidx.core.app.CoreComponentFactory.instantiateReceiver(Unknown Source:0)
	at android.app.ActivityThread.handleReceiver(ActivityThread.java:4852)
	... 9 more```

**Comment by m4teh** [comment]
Getting the same on every device startup. Pixel 8 Pro

**Comment by ImranR98** [comment]
Should be fixed (will reopen if not): https://github.com/ImranR98/Obtainium/releases/tag/v0.15.4-beta

**Comment by m4teh** [comment]
Fixed with 0.15.4-beta ❤
